config/suppressions.xml

```diff
@@ -112,5 +112,38 @@
 
   <!-- Usage of unicode characters to assert the functioning of Xpath on escape character.  -->
   <suppress id="RegexpSinglelineJava" files="[\\/]XpathQueryGeneratorTest.java"/>
+  
+  <!-- these are allowed and legacy use cases -->
+  <suppress id="noUsageOfGetFileContents()Method" files="AbstractFileSetCheckTest.java"/>
+  <suppress id="noUsageOfGetFileContents()Method" files="AbstractCheckTest.java"/>
+  <suppress id="noUsageOfGetFileContents()Method" files="CheckerTest.java"/>
+  <suppress id="noUsageOfGetFileContents()Method" files="SuppressionCommentFilter.java"/>
+  <suppress id="noUsageOfGetFileContents()Method" files="SuppressWithNearbyCommentFilter.java"/>
+  <suppress id="noUsageOfGetFileContents()Method" files="XpathQueryGenerator.java"/>
+  <suppress id="noUsageOfGetFileContents()Method" files="AbstractFileSetCheck.java"/>
```

Comment:
This is just a comment.

This is an example of using java check over reg exp.
> src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java
137:    protected final FileContents getFileContents() {

This is a method declaration and not an usage of the method.

Reply:
Very good point

Reply:
> getFileContents

Ok I dived deeper into the code and see what you mean here. From my understanding, XPath check is the strictest and the most secure one for this case. I vote for it. Do you all agree?

Reply:
> This is just a comment.
> 
> This is an example of a benefit using java check over reg exp.
> 
> > src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheck.java
> > 137:    protected final FileContents getFileContents() {
> 
> This is a method declaration and not an usage of the method.

Ok I dived deeper into the code and see what you mean here. From my understanding, XPath check is the strictest and the most secure one for this case. I vote for it. Do you all agree?

Reply:
Either java check or xpath is fine.

Reply:
> Either java check or xpath is fine.

ok seems like @romani agrees on xPath too.

Should I close this PR or amend it the XPath way?

Reply:
Go ahead and amend it. Just keep it rebased on the latest master.

Reply:
here is example of method detection by Xpath - https://github.com/checkstyle/checkstyle/pull/9939/files#diff-3a0a81e04acf0df0c9c8adefc1cae87e39da88c3141a7c131bcaf7a8ab12c62cR421

but in your case you need to make sure that classes are [AbstractCheck](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java) to show violation, all other usages should be allowed.

Reply:
@romani hey AbstractCheck has some abstract subclasses (AbstractSuperCheck, AbstractJavadocCheck, AbstractClassCouplingCheck,...)

We need to forbid the usage of getFileContents() in any classes that extend AbstractCheck and any classes that extend its abstract subclasses. Or otherwise stated, any classes that extend superclasses identified with texts "Abstract" and "Check".

Do I understand correctly?

Reply:
@haanhvu Please work on creating the xpath to match method calls we want to prevent.
Once that is working, we'll look at the areas it is flagging.